tests/test_kafka.py

```diff
@@ -0,0 +1,181 @@
+        kerberos_env.add_principals(principals)
+        kerberos_env.finalize()
+
+        service_kerberos_options = {
+            "service": {
+                "name": KAFKA_SERVICE_NAME,
+                "security": {
+                    "kerberos": {
+                        "enabled": True,
+                        "kdc_host_name": kerberos_env.get_host(),
+                        "kdc_host_port": int(kerberos_env.get_port()),
+                        "keytab_secret": kerberos_env.get_keytab_path(),
+                    }
+                }
+            }
+        }
+
+        sdk_install.uninstall(KAFKA_PACKAGE_NAME, KAFKA_SERVICE_NAME)
+        sdk_install.install(
+            KAFKA_PACKAGE_NAME,
+            KAFKA_SERVICE_NAME,
+            DEFAULT_KAFKA_TASK_COUNT,
+            additional_options=service_kerberos_options,
+            timeout_seconds=30 * 60)
+
+        yield kerberos_env
+
+    finally:
+        sdk_install.uninstall(KAFKA_PACKAGE_NAME, KAFKA_SERVICE_NAME)
+        if kerberos_env:
+            kerberos_env.cleanup()
+
+
+@pytest.fixture(scope='module', autouse=True)
+def setup_spark(kerberized_kafka):
+    try:
+        utils.require_spark()
+        yield
+    finally:
+        utils.teardown_spark()
+
+
+@pytest.mark.sanity
+@pytest.mark.skipif(not utils.kafka_enabled(), reason='KAFKA_ENABLED is false')
+def test_spark_and_kafka():
+    def producer_launched():
+        return utils.streaming_job_launched(PRODUCER_SERVICE_NAME)
+
+    def producer_started():
+        return utils.streaming_job_running(PRODUCER_SERVICE_NAME)
+
+    def kafka_broker_dns():
+        cmd = "dcos {package_name} --name={service_name} endpoints broker".format(
+            package_name=KAFKA_PACKAGE_NAME, service_name=KAFKA_SERVICE_NAME)
+        try:
+            stdout = subprocess.check_output(cmd, shell=True).decode('utf-8')
+        except Exception as e:
+            raise e("Failed to get broker endpoints")
+
+        return json.loads(stdout)["dns"][0]
+
+    kerberos_flag = "true" if KERBERIZED_KAFKA else "false"  # flag for using kerberized kafka given to app
+    stop_count = "48"  # some reasonable number
+    broker_dns = kafka_broker_dns()
+    topic = "top1"
+
+    big_file, big_file_url = "/mnt/mesos/sandbox/big.txt", "http://norvig.com/big.txt"
+
+    # arguments to the application
+    producer_args = " ".join([broker_dns, big_file, topic, kerberos_flag])
+
+    uris = "spark.mesos.uris=http://norvig.com/big.txt"
+    if KERBERIZED_KAFKA:
+        jaas_path = os.path.join(THIS_DIR, "resources", "spark-kafka-client-jaas.conf")
+        s3.upload_file(jaas_path)
+        jaas_uri = s3.s3_http_url("spark-kafka-client-jaas.conf")
+        uris += ",{}".format(jaas_uri)
+
+    common_args = [
+        "--conf", "spark.mesos.containerizer=mesos",
+        "--conf", "spark.scheduler.maxRegisteredResourcesWaitingTime=2400s",
+        "--conf", "spark.scheduler.minRegisteredResourcesRatio=1.0",
```

Comment:
What are these two configs for ("min/maxRegisteredResources")? Are they required? ^^

Reply:
They're not required, but it makes it so the driver will not start sending tasks until it has all of it's executors registered. I found this made running the job more stable.

Reply:
Ok